Expensify/react-native-onyx
# Try to synchronize operations within one update

- Held on https://github.com/Expensify/App/pull/42772 which updates the version of Onyx and includes this fix.





### Details


This PR changes how we process the `Onyx.update` operation:
1. Enqueue all the operations per individual key;
2. Group the collection-related keys into a single `mergeCollection` call;
3. Process the rest of the individual keys with `merge` calls;

### Related Issues

https://github.com/Expensify/App/issues/37560

### Automated Tests


I've added tests that cover the `Onyx.update()` with multiple merge operations of the same keys to verify that all the changes are batched into a single `mergeCollection` operation.

### Manual Tests
> [!WARNING]  
> Since this change alters one of the most common Onyx operations – we should perform a general regression testing of the system.

1. Perform a complex `Onyx.update()` operation with multiple updates of the same item in `merge` and `mergeCollection` operations.
2. Verify the collection items are updated with all the changes combined.

_Operation example ref:_ https://github.com/Expensify/react-native-onyx/blob/eb17a5dd0eba6b7b27f1e92e0fc81863091ba6c8/tests/unit/onyxTest.js#L1046-L1103

### Author Checklist


### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>
